### ISSUE TITLE
[BUG] C# / Dev Kit stops intellisense

### ISSUE THREAD

**Issue by pmahend1** [microsoft/vscode-dotnettools#1461]
### Describe the Issue

Not sure what exactly causes it. But with last 1-2 updates I am seeing this issue.
If you make typo while writing code, I have noticed intellisense stops working. Can not go to definition or see references. In the log I see conflicts between dlls which is weird because that does not happen all the time.

Only closing VS Code reopening fixes the isssue and it repeats again as soon as I write code again.

### Steps To Reproduce

1. Create dotnet maui project
2. Install C# Dev Kit, C#, .NET Maui, IntelliCode, IntelliCode for C# dev kit, .NET meteor extensions.
3. Edit any csharp file. Maybe make a mistake typing a class name.

### Expected Behavior

Intellisense should not stop
Dev Kit should not stop problem watcher

### Environment Information

MacOS 14.6.1 (23G93)
Macbook Air
Apple M2

C# VS Code log
```
Using dotnet configured on PATH
Dotnet path: /usr/local/share/dotnet/dotnet
Activating C# + C# Dev Kit + C# IntelliCode...
waiting for named pipe information from server...
[stdout] {"pipeName":"/var/folders/k7/zycjycgx4kz0ztf6kjlyggyr8zzsk7/T/6d4dddca.sock"}
received named pipe information from server
attempting to connect client to server...
client has connected to server
[Info  - 9:10:51 AM] [Program] Language server initialized
[Info  - 9:10:53 AM] [WorkspaceProjectFactoryService] Project /Users/fantom/Source/MaUI-4SEE/Grote4SEE/Grote4SEE.csproj loaded by C# Dev Kit
[Info  - 9:10:53 AM] [WorkspaceProjectFactoryService] Project /Users/fantom/Source/MaUI-4SEE/Grote4SEE/Grote4SEE.csproj loaded by C# Dev Kit
[Error - 9:13:07 AM] [LanguageServerHost] System.ArgumentException: '<Documenta58d2aca-ea3a-4efe-9bc6-394300de5b7a>' is not part of the workspace.
   at Microsoft.CodeAnalysis.Workspace.<>c.<OnDocumentClosedEx>b__213_0(Solution oldSolution, ValueTuple`4 data) in /_/src/Workspaces/Core/Portable/Workspace/Workspace_Editor.cs:line 662
   at Microsoft.CodeAnalysis.Workspace.SetCurrentSolutionAsync[TData](Boolean useAsync, TData data, Func`3 transformation, Boolean mayRaiseEvents, Action`3 onBeforeUpdate, Action`3 onAfterUpdate, CancellationToken cancellationToken) in /_/src/Workspaces/Core/Portable/Workspace/Workspace.cs:line 511
   at Roslyn.Utilities.ValueTaskExtensions.VerifyCompleted[T](ValueTask`1 task, String message) in /_/src/Workspaces/SharedUtilitiesAndExtensions/Compiler/Core/Utilities/ValueTaskExtensions.cs:line 24
   at Microsoft.CodeAnalysis.Workspace.SetCurrentSolution[TData](TData data, Func`3 transformation, Boolean mayRaiseEvents, Action`3 onBeforeUpdate, Action`3 onAfterUpdate) in /_/src/Workspaces/Core/Portable/Workspace/Workspace.cs:line 472
   at Microsoft.CodeAnalysis.Workspace.OnDocumentClosedEx(DocumentId documentId, TextLoader reloader, Boolean requireDocumentPresentAndOpen) in /_/src/Workspaces/Core/Portable/Workspace/Workspace_Editor.cs:line 650
   at Microsoft.CodeAnalysis.PdbSourceDocument.PdbSourceDocumentMetadataAsSourceFileProvider.TryRemoveDocumentFromWorkspace(MetadataAsSourceWorkspace workspace, String filePath) in /_/src/Features/Core/Portable/PdbSourceDocument/PdbSourceDocumentMetadataAsSourceFileProvider.cs:line 385
   at Microsoft.CodeAnalysis.MetadataAsSource.MetadataAsSourceFileService.TryRemoveDocumentFromWorkspace(String filePath) in /_/src/Features/Core/Portable/MetadataAsSource/MetadataAsSourceFileService.cs:line 203
   at Microsoft.CodeAnalysis.LanguageServer.LspMiscellaneousFilesWorkspace.TryRemoveMiscellaneousDocument(Uri uri, Boolean removeFromMetadataWorkspace) in /_/src/LanguageServer/Protocol/Workspaces/LspMiscellaneousFilesWorkspace.cs:line 80
   at Microsoft.CodeAnalysis.LanguageServer.LspWorkspaceManager.StopTrackingAsync(Uri uri, CancellationToken cancellationToken) in /_/src/LanguageServer/Protocol/Workspaces/LspWorkspaceManager.cs:line 199
   at Microsoft.CodeAnalysis.LanguageServer.Handler.DocumentChanges.DidCloseHandler.HandleNotificationAsync(DidCloseTextDocumentParams request, RequestContext context, CancellationToken cancellationToken) in /_/src/LanguageServer/Protocol/Handler/DocumentChanges/DidCloseHandler.cs:line 36
   at Microsoft.CommonLanguageServerProtocol.Framework.QueueItem`1.StartRequestAsync[TRequest,TResponse](TRequest request, TRequestContext context, IMethodHandler handler, String language, CancellationToken cancellationToken) in /_/src/LanguageServer/Microsoft.CommonLanguageServerProtocol.Framework/QueueItem.cs:line 199
[Error - 9:13:07 AM] [LanguageServerHost] System.ArgumentException: '<Documenta58d2aca-ea3a-4efe-9bc6-394300de5b7a>' is not part of the workspace.
   at Microsoft.CodeAnalysis.Workspace.<>c.<OnDocumentClosedEx>b__213_0(Solution oldSolution, ValueTuple`4 data) in /_/src/Workspaces/Core/Portable/Workspace/Workspace_Editor.cs:line 662
   at Microsoft.CodeAnalysis.Workspace.SetCurrentSolutionAsync[TData](Boolean useAsync, TData data, Func`3 transformation, Boolean mayRaiseEvents, Action`3 onBeforeUpdate, Action`3 onAfterUpdate, CancellationToken cancellationToken) in /_/src/Workspaces/Core/Portable/Workspace/Workspace.cs:line 511
   at Roslyn.Utilities.ValueTaskExtensions.VerifyCompleted[T](ValueTask`1 task, String message) in /_/src/Workspaces/SharedUtilitiesAndExtensions/Compiler/Core/Utilities/ValueTaskExtensions.cs:line 24
   at Microsoft.CodeAnalysis.Workspace.SetCurrentSolution[TData](TData data, Func`3 transformation, Boolean mayRaiseEvents, Action`3 onBeforeUpdate, Action`3 onAfterUpdate) in /_/src/Workspaces/Core/Portable/Workspace/Workspace.cs:line 472
   at Microsoft.CodeAnalysis.Workspace.OnDocumentClosedEx(DocumentId documentId, TextLoader reloader, Boolean requireDocumentPresentAndOpen) in /_/src/Workspaces/Core/Portable/Workspace/Workspace_Editor.cs:line 650
   at Microsoft.CodeAnalysis.PdbSourceDocument.PdbSourceDocumentMetadataAsSourceFileProvider.TryRemoveDocumentFromWorkspace(MetadataAsSourceWorkspace workspace, String filePath) in /_/src/Features/Core/Portable/PdbSourceDocument/PdbSourceDocumentMetadataAsSourceFileProvider.cs:line 385
   at Microsoft.CodeAnalysis.MetadataAsSource.MetadataAsSourceFileService.TryRemoveDocumentFromWorkspace(String filePath) in /_/src/Features/Core/Portable/MetadataAsSource/MetadataAsSourceFileService.cs:line 203
   at Microsoft.CodeAnalysis.LanguageServer.LspMiscellaneousFilesWorkspace.TryRemoveMiscellaneousDocument(Uri uri, Boolean removeFromMetadataWorkspace) in /_/src/LanguageServer/Protocol/Workspaces/LspMiscellaneousFilesWorkspace.cs:line 80
   at Microsoft.CodeAnalysis.LanguageServer.LspWorkspaceManager.StopTrackingAsync(Uri uri, CancellationToken cancellationToken) in /_/src/LanguageServer/Protocol/Workspaces/LspWorkspaceManager.cs:line 199
   at Microsoft.CodeAnalysis.LanguageServer.Handler.DocumentChanges.DidCloseHandler.HandleNotificationAsync(DidCloseTextDocumentParams request, RequestContext context, CancellationToken cancellationToken) in /_/src/LanguageServer/Protocol/Handler/DocumentChanges/DidCloseHandler.cs:line 36
   at Microsoft.CommonLanguageServerProtocol.Framework.QueueItem`1.StartRequestAsync[TRequest,TResponse](TRequest request, TRequestContext context, IMethodHandler handler, String language, CancellationToken cancellationToken) in /_/src/LanguageServer/Microsoft.CommonLanguageServerProtocol.Framework/QueueItem.cs:line 199
   at Microsoft.CommonLanguageServerProtocol.Framework.QueueItem`1.StartRequestAsync[TRequest,TResponse](TRequest request, TRequestContext context, IMethodHandler handler, String language, CancellationToken cancellationToken) in /_/src/LanguageServer/Microsoft.CommonLanguageServerProtocol.Framework/QueueItem.cs:line 241
   at Microsoft.CodeAnalysis.LanguageServer.RoslynRequestExecutionQueue.WrapStartRequestTaskAsync(Task nonMutatingRequestTask, Boolean rethrowExceptions) in /_/src/LanguageServer/Protocol/RoslynRequestExecutionQueue.cs:line 40
   at Microsoft.CommonLanguageServerProtocol.Framework.RequestExecutionQueue`1.ProcessQueueCoreAsync[TRequest,TResponse](IQueueItem`1 work, IMethodHandler handler, RequestHandlerMetadata metadata, ConcurrentDictionary`2 concurrentlyExecutingTasks, CancellationTokenSource currentWorkCts, CancellationToken cancellationToken) in /_/src/LanguageServer/Microsoft.CommonLanguageServerProtocol.Framework/RequestExecutionQueue.cs:line 349
   at Microsoft.CommonLanguageServerProtocol.Framework.RequestExecutionQueue`1.InvokeProcessCoreAsync(IQueueItem`1 work, RequestHandlerMetadata metadata, IMethodHandler handler, MethodInfo methodInfo, ConcurrentDictionary`2 concurrentlyExecutingTasks, CancellationTokenSource currentWorkCts, CancellationToken cancellationToken) in /_/src/LanguageServer/Microsoft.CommonLanguageServerProtocol.Framework/RequestExecutionQueue.cs:line 302
   at Microsoft.CommonLanguageServerProtocol.Framework.RequestExecutionQueue`1.ProcessQueueAsync() in /_/src/LanguageServer/Microsoft.CommonLanguageServerProtocol.Framework/RequestExecutionQueue.cs:line 254
[Info  - 9:13:07 AM] Error processing queue, shutting down
[Error - 9:13:07 AM] Workspace diagnostic pull failed.
Canceled: Canceled
	at a.handleFailedRequest (/Users/fantom/.vscode/extensions/ms-dotnettools.csharp-2.45.20-darwin-arm64/dist/extension.js:2:2137421)
	at a.handleFailedRequest (/Users/fantom/.vscode/extensions/ms-dotnettools.csharp-2.45.20-darwin-arm64/dist/extension.js:2:1253165)
	at /Users/fantom/.vscode/extensions/ms-dotnettools.csharp-2.45.20-darwin-arm64/dist/extension.js:2:2174833
	at async g.pullWorkspaceAsync (/Users/fantom/.vscode/extensions/ms-dotnettools.csharp-2.45.20-darwin-arm64/dist/extension.js:2:2172141)
[Info  - 9:13:07 AM] Connection to server got closed. Server will restart.
true
Using dotnet configured on PATH
Dotnet path: /usr/local/share/dotnet/dotnet
Activating C# + C# Dev Kit + C# IntelliCode...
waiting for named pipe information from server...
[stdout] fail: Microsoft.CodeAnalysis.LanguageServer.ExportProviderBuilder[0]
      /Users/fantom/.vscode/extensions/ms-dotnettools.csharp-2.45.20-darwin-arm64/.xamlTools/Microsoft.VisualStudio.DesignTools.CodeAnalysis.dll with assembly name Microsoft.VisualStudio.DesignTools.CodeAnalysis, Version=17.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a conflicts with extension loaded from /Users/fantom/.vscode/extensions/ms-dotnettools.csharp-2.45.20-darwin-arm64/.xamlTools
fail: Microsoft.CodeAnalysis.LanguageServer.ExportProviderBuilder[0]
      /Users/fantom/.vscode/extensions/ms-dotnettools.csharp-2.45.20-darwin-arm64/.xamlTools/Microsoft.VisualStudio.DesignTools.CodeAnalysis.Diagnostics.dll with assembly name Microsoft.VisualStudio.DesignTools.CodeAnalysis.Diagnostics, Version=17.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a conflicts with extension loaded from /Users/fantom/.vscode/extensions/ms-dotnettools.csharp-2.45.20-darwin-arm64/.xamlTools
[stdout] {"pipeName":"/var/folders/k7/zycjycgx4kz0ztf6kjlyggyr8zzsk7/T/2f8856e7.sock"}
received named pipe information from server
attempting to connect client to server...
client has connected to server
[Info  - 9:13:08 AM] [Program] Language server initialized
```

C# LSP Trace logs
```log
Extension vscode.bat does not contribute csharpExtensionLoadPaths
Extension vscode.clojure does not contribute csharpExtensionLoadPaths
Extension vscode.coffeescript does not contribute csharpExtensionLoadPaths
Extension vscode.configuration-editing does not contribute csharpExtensionLoadPaths
Extension vscode.cpp does not contribute csharpExtensionLoadPaths
Extension vscode.csharp does not contribute csharpExtensionLoadPaths
Extension vscode.css does not contribute csharpExtensionLoadPaths
Extension vscode.css-language-features does not contribute csharpExtensionLoadPaths
Extension vscode.dart does not contribute csharpExtensionLoadPaths
Extension vscode.debug-auto-launch does not contribute csharpExtensionLoadPaths
Extension vscode.debug-server-ready does not contribute csharpExtensionLoadPaths
Extension vscode.diff does not contribute csharpExtensionLoadPaths
Extension vscode.docker does not contribute csharpExtensionLoadPaths
Extension vscode.emmet does not contribute csharpExtensionLoadPaths
Extension vscode.extension-editing does not contribute csharpExtensionLoadPaths
Extension vscode.fsharp does not contribute csharpExtensionLoadPaths
Extension vscode.git does not contribute csharpExtensionLoadPaths
Extension vscode.git-base does not contribute csharpExtensionLoadPaths
Extension vscode.github does not contribute csharpExtensionLoadPaths
Extension vscode.github-authentication does not contribute csharpExtensionLoadPaths
Extension vscode.go does not contribute csharpExtensionLoadPaths
Extension vscode.groovy does not contribute csharpExtensionLoadPaths
Extension vscode.grunt does not contribute csharpExtensionLoadPaths
Extension vscode.gulp does not contribute csharpExtensionLoadPaths
Extension vscode.handlebars does not contribute csharpExtensionLoadPaths
Extension vscode.hlsl does not contribute csharpExtensionLoadPaths
Extension vscode.html does not contribute csharpExtensionLoadPaths
Extension vscode.html-language-features does not contribute csharpExtensionLoadPaths
Extension vscode.ini does not contribute csharpExtensionLoadPaths
Extension vscode.ipynb does not contribute csharpExtensionLoadPaths
Extension vscode.jake does not contribute csharpExtensionLoadPaths
Extension vscode.java does not contribute csharpExtensionLoadPaths
Extension vscode.javascript does not contribute csharpExtensionLoadPaths
Extension vscode.json does not contribute csharpExtensionLoadPaths
Extension vscode.json-language-features does not contribute csharpExtensionLoadPaths
Extension vscode.julia does not contribute csharpExtensionLoadPaths
Extension vscode.latex does not contribute csharpExtensionLoadPaths
Extension vscode.less does not contribute csharpExtensionLoadPaths
Extension vscode.log does not contribute csharpExtensionLoadPaths
Extension vscode.lua does not contribute csharpExtensionLoadPaths
Extension vscode.make does not contribute csharpExtensionLoadPaths
Extension vscode.markdown does not contribute csharpExtensionLoadPaths
Extension vscode.markdown-language-features does not contribute csharpExtensionLoadPaths
Extension vscode.markdown-math does not contribute csharpExtensionLoadPaths
Extension vscode.media-preview does not contribute csharpExtensionLoadPaths
Extension vscode.merge-conflict does not contribute csharpExtensionLoadPaths
Extension vscode.microsoft-authentication does not contribute csharpExtensionLoadPaths
Extension ms-vscode.js-debug does not contribute csharpExtensionLoadPaths
Extension ms-vscode.js-debug-companion does not contribute csharpExtensionLoadPaths
Extension ms-vscode.vscode-js-profile-table does not contribute csharpExtensionLoadPaths
Extension vscode.builtin-notebook-renderers does not contribute csharpExtensionLoadPaths
Extension vscode.npm does not contribute csharpExtensionLoadPaths
Extension vscode.objective-c does not contribute csharpExtensionLoadPaths
Extension vscode.perl does not contribute csharpExtensionLoadPaths
Extension vscode.php does not contribute csharpExtensionLoadPaths
Extension vscode.php-language-features does not contribute csharpExtensionLoadPaths
Extension vscode.powershell does not contribute csharpExtensionLoadPaths
Extension vscode.pug does not contribute csharpExtensionLoadPaths
Extension vscode.python does not contribute csharpExtensionLoadPaths
Extension vscode.r does not contribute csharpExtensionLoadPaths
Extension vscode.razor does not contribute csharpExtensionLoadPaths
Extension vscode.references-view does not contribute csharpExtensionLoadPaths
Extension vscode.restructuredtext does not contribute csharpExtensionLoadPaths
Extension vscode.ruby does not contribute csharpExtensionLoadPaths
Extension vscode.rust does not contribute csharpExtensionLoadPaths
Extension vscode.scss does not contribute csharpExtensionLoadPaths
Extension vscode.search-result does not contribute csharpExtensionLoadPaths
Extension vscode.shaderlab does not contribute csharpExtensionLoadPaths
Extension vscode.shellscript does not contribute csharpExtensionLoadPaths
Extension vscode.simple-browser does not contribute csharpExtensionLoadPaths
Extension vscode.sql does not contribute csharpExtensionLoadPaths
Extension vscode.swift does not contribute csharpExtensionLoadPaths
Extension vscode.theme-abyss does not contribute csharpExtensionLoadPaths
Extension vscode.theme-defaults does not contribute csharpExtensionLoadPaths
Extension vscode.theme-kimbie-dark does not contribute csharpExtensionLoadPaths
Extension vscode.theme-monokai does not contribute csharpExtensionLoadPaths
Extension vscode.theme-monokai-dimmed does not contribute csharpExtensionLoadPaths
Extension vscode.theme-quietlight does not contribute csharpExtensionLoadPaths
Extension vscode.theme-red does not contribute csharpExtensionLoadPaths
Extension vscode.vscode-theme-seti does not contribute csharpExtensionLoadPaths
Extension vscode.theme-solarized-dark does not contribute csharpExtensionLoadPaths
Extension vscode.theme-solarized-light does not contribute csharpExtensionLoadPaths
Extension vscode.theme-tomorrow-night-blue does not contribute csharpExtensionLoadPaths
Extension vscode.tunnel-forwarding does not contribute csharpExtensionLoadPaths
Extension vscode.typescript does not contribute csharpExtensionLoadPaths
Extension vscode.typescript-language-features does not contribute csharpExtensionLoadPaths
Extension vscode.vb does not contribute csharpExtensionLoadPaths
Extension vscode.xml does not contribute csharpExtensionLoadPaths
Extension vscode.yaml does not contribute csharpExtensionLoadPaths
Extension eamodio.gitlens does not contribute csharpExtensionLoadPaths
Extension Gruntfuggly.todo-tree does not contribute csharpExtensionLoadPaths
Extension just-seba.vscode-code-butler does not contribute csharpExtensionLoadPaths
Extension ms-dotnettools.csdevkit does not contribute csharpExtensionLoadPaths
Extension ms-dotnettools.csharp does not contribute csharpExtensionLoadPaths
Extension ms-dotnettools.vscode-dotnet-runtime does not contribute csharpExtensionLoadPaths
Extension ms-dotnettools.vscodeintellicode-csharp does not contribute csharpExtensionLoadPaths
Extension nromanov.dotnet-meteor does not contribute csharpExtensionLoadPaths
Extension PrateekMahendrakar.prettyxml does not contribute csharpExtensionLoadPaths
Extension PrateekMahendrakar.resxpress does not contribute csharpExtensionLoadPaths
Extension VisualStudioExptTeam.intellicode-api-usage-examples does not contribute csharpExtensionLoadPaths
Extension VisualStudioExptTeam.vscodeintellicode does not contribute csharpExtensionLoadPaths
```

**Comment by CerosDev** [comment]
Might be related to https://github.com/dotnet/vscode-csharp/issues/7514 and https://github.com/dotnet/vscode-csharp/issues/7534

**Comment by dibarbet** [comment]
Duplicate of https://github.com/dotnet/vscode-csharp/issues/7514.  Should be fixed in 2.45.25 (just released), see https://github.com/dotnet/vscode-csharp/issues/7514#issuecomment-2347181280
